nsgcli/grok.py

```diff
@@ -92,10 +94,49 @@ def do_parse(self, _):
         if not txt:
             self.read_stdin(pattern)
         else:
-            self.call_grok_api(txt, pattern)
+            self.call_grok_api("log", txt, pattern)
+
+    ##########################################################################################
+    def do_text(self, _):
+        """
+        Parse text line with grok patterns
+
+        text [--text <text>] [--pattern <pattern>]
+
+        Parameters:
+            --text Text to be parsed, read stdin if not specified
+            --pattern Grok pattern to be applied to input text.
+
+        Examples:
+
+            text --text "Hello world of Grok" --pattern "Hello world of %{WORD:world_name}"
+
+            text --pattern "Hello world of %{WORD:world_name}" <<<
+            "Hello world of Grok"
+        """
+        try:
+            opts, args = getopt.getopt(sys.argv[3:],
+                                       't:p:',
+                                       ['text=', 'pattern='])
+        except getopt.GetoptError as ex:
+            print('UNKNOWN: Invalid Argument:' + str(ex))
+            raise InvalidArgsException
+
+        txt = None
+        pattern = None
+        for opt, arg in opts:
+            if opt in ['-t', '--text']:
```

Comment:
this looks strange. Why reach for `sys.argv` again if you should be getting parameters as a second parameter in the call to `do_log()` and `do_text()` ?

also I do not know how do I feel about using getopt inside of the command (i.e. `text --pattern ... --text`) . This contradicts general design of the CLI in nsgcli that is based around Python module Cmd. Arguments of other commands are positional and are not preceded with "--word" .